### ISSUE TITLE
GCC Cross Compiler Strip

### ISSUE THREAD

**Issue by bfrog** [NixOS/nixpkgs#109295]
**Describe the bug**
Currently cross compiler toolchains using gcc set dontStrip = true.

**To Reproduce**
1. build any software with crossSystem set to a different target arch, see gcc has -debug added, and all debug symbols

**Expected behavior**
Only build -debug variant when override is set

**Additional context**
The root of the problem seems to stem from strip.sh using the same strip binary for both libraries and binaries, where in a cross compiler toolchain what probably should happen is the target strip should be used against libraries, and the host strip should be used against binaries.

**Notify maintainers**
@Synthetica9

**Comment by bfrog** [comment]
I'm happy to sponsor a fix for this if its reasonable!

**Comment by bfrog** [comment]
@matthewbauer thought you might be interested in this one

**Comment by matthewbauer** [comment]
Yeah! So I can definitely merge https://github.com/NixOS/nixpkgs/pull/108060, but you're right that it doesn't actually strip true cross-compilation.

@Ericson2314 originally introduced the TARGET_STRIP & STRIP distinction. It's possible we can just set "stripped" always to true, but I would want to make sure we aren't running "strip" on the wrong things.

**Comment by Ericson2314** [comment]
Yeah `TARGET_STRIP` & `STRIP` was something I made for this purpose, but never found time to put it at the test. At this point, I rather invest in the real solution which is building GCC itself and the target libraries separately. This will improve and simplify things on so many levels.

**Comment by bfrog** [comment]
Not stale

**Comment by teburd** [comment]
Not Stale, Still Relevant

**Comment by Mindavi** [comment]
I thought this was done to prevent broken binaries, since (IIRC correctly) ARM binaries were sometimes broken when stripped.  But maybe the situation has improved?

**Comment by minijackson** [comment]
So I tried cross-compiling a minimal all-stripped armv7l NixOS image and tested it a bit on a Raspberry PI 2, and it boots and works without any issue. Note that comments in compilers' `default.nix` state:

> Strip kills static libs of other archs (hence no cross)

I'm not sure how to test that.

**Comment by trofi** [comment]
Possible fix for cross-compiler stripping: https://github.com/NixOS/nixpkgs/pull/182513
